src/Backend/Modules/Blog/Engine/Model.php

```diff
@@ -1113,67 +1101,50 @@ public static function updateCategory(array $item, array $meta = null): int
-     *
-     * @param array $item The new data.
-     *
-     * @return int
-     */
-    public static function updateComment(array $item): int
+    public static function updateComment(array $item)
     {
-        // update category
-        return BackendModel::getContainer()->get('database')->update(
-            'blog_comments',
-            $item,
-            'id = ?',
-            [(int) $item['id']]
+        $entityManager = BackendModel::get('doctrine.orm.default_entity_manager');
+        $repository = $entityManager->getRepository(Comment::class);
+        $comment = $repository->find($item['id']);
+
+        if (!$comment instanceof Comment) {
+            return;
+        }
+
+        $comment->update(
+            $item['author'],
+            $item['email'],
+            $item['website'],
+            $item['text'],
+            $comment->getType(),
+            $item['status'],
+            $comment->getData()
         );
+
+        $entityManager->flush($comment);
     }
 
-    /**
-     * Updates one or more comments' status
-     *
-     * @param array  $ids    The id(s) of the comment(s) to change the status for.
-     * @param string $status The new status.
-     */
-    public static function updateCommentStatuses(array $ids, string $status): void
-    {
-        // make sure $ids is an array
-        $ids = (array) $ids;
+    public static function updateCommentStatuses(array $ids, string $status): void{
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
+            ->getRepository(Comment::class);
 
-        // loop and cast to integers
-        foreach ($ids as &$id) {
-            $id = (int) $id;
-        }
+        $comments = $repository->findById($ids);
 
-        // create an array with an equal amount of questionmarks as ids provided
-        $idPlaceHolders = array_fill(0, count($ids), '?');
+        if (empty($comments)) {
+            return;
+        }
 
-        // get the items and their languages
-        $items = (array) BackendModel::getContainer()->get('database')->getPairs(
-            'SELECT i.post_id, i.language
-             FROM blog_comments AS i
-             WHERE i.id IN (' . implode(', ', $idPlaceHolders) . ')',
-            $ids,
-            'post_id'
-        );
+        $postIdsToRecalculate = [];
+        foreach ($comments as $comment) {
+            $postIdsToRecalculate[] = $comment->getPostId();
+        }
 
-        // only proceed if there are items
-        if (!empty($items)) {
-            // get the ids
-            $itemIds = array_keys($items);
-
-            // update records
-            BackendModel::getContainer()->get('database')->execute(
-                'UPDATE blog_comments
-                 SET status = ?
-                 WHERE id IN (' . implode(', ', $idPlaceHolders) . ')',
-                array_merge([(string) $status], $ids)
-            );
+        $repository->updateMultipleStatusById($ids, $status);
```

Comment:
can you just do before this, then we don't need that if below anymore
```php
if (empty($postIdsToRecalculate)) {
    return;
}
```

Reply:
Fixed in 18adc30c0965878417a31b49881be34846e4911a